core/model.py

```diff
@@ -0,0 +1,230 @@
+        self._set_value_from_data(data)
+
+    def _set_value_from_data(self, data: dict):
+        for key in self._fields:
+            setattr(self, key, data.get(key))
+
+    def _get_data_load_object(self, data: dict) -> dict:
+        data_load_object = {}
+        if not data:
+            return data_load_object
+        for key in data:
+            if key in _filter_out_load_object_key or key not in data or key not in self._fields:
+                continue
+
+            value = data[key]
+            if key in _prefilled_fields: # is _id
+                data_load_object[key] = value
+            elif self._is_reference_field(key): # is one2many reference
+                elements = []
+                reference_model = self._get_reference_model_of_field(key)
+                for element in value:
+                    elements.append(reference_model.get(element))
+                data_load_object[key] = elements
+            elif self.__annotations__[key].__base__ is Model: # is many2one reference
+                data_load_object[key] = self.__annotations__[key].get(value)
+            elif self.__annotations__[key].__base__ is Enum: # is enum
+                data_load_object[key] = self.__annotations__[key](value)
+            else: # is general type
+                data_load_object[key] = value
+        return data_load_object
+
+    def _get_data_store_object(self) -> dict:
+        data_store_object = {}
+        for key in self.__dict__.keys():
+            if key in _filter_out_store_object_key:
+                continue
+
+            value = self.__dict__[key]
+            if value is None:
+                    continue
+            if self._is_reference_field(key): # is one2many reference
+                element_ids = []
+                for element in value:
+                    if not element._is_saved():
+                        element.save()
+                    element_ids.append(element._id)
+                data_store_object[key] = element_ids
+            elif isinstance(value, Model): # is many2one reference
+                if not value._is_saved():
+                    value.save()
+                data_store_object[key] = value._id
+            elif isinstance(value, Enum): # is enum
+                data_store_object[key] = value.value
+            else: # is general type
+                data_store_object[key] = value
+        return data_store_object
+
+    def _is_saved(self) -> bool:
+        return hasattr(self, '_id') and self._id
+    
+    def _is_reference_field(self, field: str) -> bool:
+        return field in self.__annotations__ and len(get_args(self.__annotations__[field])) == 1 and isinstance(get_args(self.__annotations__[field])[0](), Model)
+    
+    def _get_reference_model_of_field(self, field: str):
+        return get_args(self.__annotations__[field])[0]
+
+    @classmethod
+    def _get_fields(cls):
+        return list(cls.__annotations__.keys()) + _prefilled_fields
```

Comment:
Hm, wouldn't this duplicate `_id`, `created_at` and `updated_at` fields? Aren't these values in both `__anotations__` and in `_prefilled_fields`?

Reply:
Somehow in `cls.__annotations__.keys()` does not contain attributes in `Model` main class, it only contain only their attributes in inherited class.

Reply:
Is `_prefilled_fields` then even needed at all? It does not seem used besides `_get_data_load_object`, but the `if key in _prefilled_fields: # is _id` seems redundant.

Reply:
Yes. I think we still need `_prefilled_fields`. The main reason is the field in parent class (`Model`) will not be appeared in `__annotations__` in subclass. So if we don't check `if key in _prefilled_fields` it will type to get refilled fields' type from `__annotations__` which does not exist and raise an exception.

Reply:
Ah, so `_prefilled_fields` mean "default fields that appear in every object". Anyway, maybe wrap it in a `set()` in case someone duplicates some keys in annotations?

`return list(set( annotations + prefilled ))` ?